### ISSUE TITLE
Some questions about the code

### ISSUE THREAD

**Issue by Sunny-gyx** [hhrd9/regmixmatch#1]
Hello, I have a few questions about the code of the algorithm.First of all,

![Image](https://github.com/user-attachments/assets/cfb39b4b-0949-4184-8749-33e47d4b59bd)

![Image](https://github.com/user-attachments/assets/369adb22-4b32-46d1-b81a-461dc1b0f9ad)  according to your code, the return values of the function "consistency loss" do not seem to match with the values obtained when calling the function?for example, pseudo labels should be the fifth value of the return values, however , when the function is called in the function"train", pseudo labels seem to be the third value? I am also confused with the values"prob_w" and "pseudo_lb". Could you please explain the difference? 

Moreover,I don't understand the calculation of the l2 loss either, which is also part of the cam algorithm. Why do you mutiply the l2-loss by the cube of "max_probs" of the low-conf data? I have examined the paper for several times and haven't found the origin.

![Image](https://github.com/user-attachments/assets/8165d2d0-3109-414b-80c1-78bb56a0c3b7)

 Lastly, I am curious that why the conf-data in our code concatenates the labeled data and truly high-conf data? It seems that the high-conf data mentioned in the paper only refers to unlabeled data? I wonder if  it is a mistake.
I am looking forward to your timely reply.Thanks in advance.Wish you all the best. @hhrd9

**Comment by hhrd9** [comment]
In the calculation of the consistency_loss function, the "pseudo_label" is obtained by applying softmax to "logits_w", which represents the predicted probability "prob_w" of  data. In fact, the augmax of "prob_w"("max_id") is the actual pseudo-labels. Therefore, the "max_id" in the function's return is represented as "pseudo_lb" in the main function. I believe I should use another appropriate term to replace "pseudo_label" in the consistency_loss function.

The CAM algorithm is used to calculate the loss for low-confidence samples. The labels for these samples are usually unreliable, so multiplying their loss by "max_probs" implements a form of confidence weighting. I consider this a minor adjustment, which is why it was not explicitly included in the paper.

In essence, labeled data can be considered as the highest-confidence unlabeled data. Therefore, I concatenated it with high-confidence unlabeled data.
